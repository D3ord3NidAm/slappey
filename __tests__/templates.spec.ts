import * as templates from "../src/templates/templates";

describe("Templates", () => {
  it("should match envTemplate", () => {
    expect(templates.getEnvTemplate("123", "123")).toMatchSnapshot();
  });
  it("should match getMainFile", () => {
    expect(templates.getMainFile()).toMatchSnapshot();
  });
  it("should match getMainFileTS", () => {
    expect(templates.getMainFileTS()).toMatchSnapshot();
  });
  it("should match getTypescriptBotFile", () => {
    expect(templates.getTypescriptBotFile()).toMatchSnapshot();
  });
  it("should match getRegistryFileTS", () => {
    expect(templates.getRegistryFileTS()).toMatchSnapshot();
  });
  it("should match getRegistryFile", () => {
    expect(templates.getRegistryFile()).toMatchSnapshot();
  });
  it("should match getBaseCommand", () => {
    expect(templates.getBaseCommand()).toMatchSnapshot();
  });
  it("should match getBaseCommandTS", () => {
    expect(templates.getBaseCommandTS()).toMatchSnapshot();
  });
  it("should match getBaseEvent", () => {
    expect(templates.getBaseEvent()).toMatchSnapshot();
  });
  it("should match getBaseEventTS", () => {
    expect(templates.getBaseEventTS()).toMatchSnapshot();
  });
  it("should match getReadyEvent", () => {
    expect(templates.getReadyEvent()).toMatchSnapshot();
  });
  it("should match getReadyEventTS", () => {
    expect(templates.getReadyEventTS()).toMatchSnapshot();
  });
  it("should match getMessageEvent", () => {
    expect(templates.getMessageEvent()).toMatchSnapshot();
  });
  it("should match getMessageEventTS", () => {
    expect(templates.getMessageEventTS()).toMatchSnapshot();
  });
  it("should match getTestCommand", () => {
    expect(templates.getTestCommand()).toMatchSnapshot();
  });
  it("should match getTestCommandTS", () => {
    expect(templates.getTestCommandTS()).toMatchSnapshot();
  });
  it("should match getCommandTemplate", () => {
    expect(templates.getCommandTemplate("Ban", "mod")).toMatchSnapshot();
  });
  it("should match getCommandTemplateTS", () => {
    expect(templates.getCommandTemplateTS("Kick", "mod")).toMatchSnapshot();
  });
});
