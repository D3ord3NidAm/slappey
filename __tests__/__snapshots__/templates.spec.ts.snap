// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates should match envTemplate 1`] = `
"DISCORD_BOT_TOKEN=123
DISCORD_BOT_PREFIX=123"
`;

exports[`Templates should match getBaseCommand 1`] = `
"module.exports = class BaseCommand {
  constructor(name, category, aliases) {
    this.name = name;
    this.category = category;
    this.aliases = aliases;
  }
}"
`;

exports[`Templates should match getBaseCommandTS 1`] = `
"
import { Message } from 'discord.js';
import DiscordClient from '../../client/client';

export default abstract class BaseCommand {
  constructor(private name: string, private category: string, private aliases: Array<string>) {}

  getName(): string { return this.name; }
  getCategory(): string { return this.category; }
  getAliases(): Array<string> { return this.aliases; }

  abstract async run(client: DiscordClient, message: Message, args: Array<string> | null): Promise<void>;
}"
`;

exports[`Templates should match getBaseEvent 1`] = `
"module.exports = class BaseEvent {
  constructor(name) {
    this.name = name;
  }
}"
`;

exports[`Templates should match getBaseEventTS 1`] = `
"
import DiscordClient from '../../client/client';

export default abstract class BaseEvent {
  constructor(private name: string) { }

  getName(): string { return this.name; }
  abstract run(client: DiscordClient, ...args: any): void;
}
"
`;

exports[`Templates should match getCommandTemplate 1`] = `
"const BaseCommand = require('../../utils/structures/BaseCommand');

module.exports = class BanCommand extends BaseCommand {
  constructor() {
    super('Ban', 'mod', []);
  }

  run(client, message, args) {
    message.channel.send('Ban command works');
  }
}"
`;

exports[`Templates should match getCommandTemplateTS 1`] = `
"import { Message } from 'discord.js';
import BaseCommand from '../../utils/structures/BaseCommand';
import DiscordClient from '../../client/client';

export default class KickCommand extends BaseCommand {
  constructor() {
    super('Kick', 'mod', []);
  }

  async run(client: DiscordClient, message: Message, args: Array<string>) {
    message.channel.send('Kick command works');
  }
}"
`;

exports[`Templates should match getMainFile 1`] = `
"require('dotenv').config();
const { Client } = require('discord.js');
const { registerCommands, registerEvents } = require('./utils/registry');
const client = new Client();
const config = require('../slappey.json');

(async () => {
  client.commands = new Map();
  client.events = new Map();
  client.prefix = process.env.DISCORD_BOT_PREFIX;
  await registerCommands(client, '../commands');
  await registerEvents(client, '../events');
  await client.login(config.token);
})();

"
`;

exports[`Templates should match getMainFileTS 1`] = `
"import { config } from 'dotenv';
config();
import { registerCommands, registerEvents } from './utils/registry';
import DiscordClient from './client/client';
const client = new DiscordClient({});
const config = require('../slappey.json');

(async () => {
  client.prefix = process.env.DISCORD_BOT_PREFIX || client.prefix;
  await registerCommands(client, '../commands');
  await registerEvents(client, '../events');
  await client.login(config.token);
})();

"
`;

exports[`Templates should match getMessageEvent 1`] = `
"const BaseEvent = require('../../utils/structures/BaseEvent');

module.exports = class MessageEvent extends BaseEvent {
  constructor() {
    super('message');
  }
  
  async run(client, message) {
    if (message.author.bot) return;
    if (message.content.startsWith(client.prefix)) {
      const [cmdName, ...cmdArgs] = message.content
      .slice(client.prefix.length)
      .trim()
      .split(/\\\\s+/);
      const command = client.commands.get(cmdName);
      if (command) {
        command.run(client, message, cmdArgs);
      }
    }
  }
}"
`;

exports[`Templates should match getMessageEventTS 1`] = `
"import BaseEvent from '../../utils/structures/BaseEvent';
import { Message } from 'discord.js';
import DiscordClient from '../../client/client';

export default class MessageEvent extends BaseEvent {
  constructor() {
    super('message');
  }

  async run(client: DiscordClient, message: Message) {
    if (message.author.bot) return;
    if (message.content.startsWith(client.prefix)) {
      const [cmdName, ...cmdArgs] = message.content
        .slice(client.prefix.length)
        .trim()
        .split(/\\\\s+/);
      const command = client.commands.get(cmdName);
      if (command) {
        command.run(client, message, cmdArgs);
      }
    }
  }
}"
`;

exports[`Templates should match getReadyEvent 1`] = `
"const BaseEvent = require('../../utils/structures/BaseEvent');

module.exports = class ReadyEvent extends BaseEvent {
  constructor() {
    super('ready');
  }
  async run (client) {
    console.log(client.user.tag + ' has logged in.');
  }
}"
`;

exports[`Templates should match getReadyEventTS 1`] = `
"import BaseEvent from '../../utils/structures/BaseEvent';
import DiscordClient from '../../client/client';

export default class ReadyEvent extends BaseEvent {
  constructor() {
    super('ready');
  }
  async run (client: DiscordClient) {
    console.log('Bot has logged in.');
  }
}"
`;

exports[`Templates should match getRegistryFile 1`] = `
"
const path = require('path');
const fs = require('fs').promises;
const BaseCommand = require('./structures/BaseCommand');
const BaseEvent = require('./structures/BaseEvent');

async function registerCommands(client, dir = '') {
  const filePath = path.join(__dirname, dir);
  const files = await fs.readdir(filePath);
  for (const file of files) {
    const stat = await fs.lstat(path.join(filePath, file));
    if (stat.isDirectory()) registerCommands(client, path.join(dir, file));
    if (file.endsWith('.js')) {
      const Command = require(path.join(filePath, file));
      if (Command.prototype instanceof BaseCommand) {
        const cmd = new Command();
        client.commands.set(cmd.name, cmd);
        cmd.aliases.forEach((alias) => {
          client.commands.set(alias, cmd);
        });
      }
    }
  }
}

async function registerEvents(client, dir = '') {
  const filePath = path.join(__dirname, dir);
  const files = await fs.readdir(filePath);
  for (const file of files) {
    const stat = await fs.lstat(path.join(filePath, file));
    if (stat.isDirectory()) registerEvents(client, path.join(dir, file));
    if (file.endsWith('.js')) {
      const Event = require(path.join(filePath, file));
      if (Event.prototype instanceof BaseEvent) {
        const event = new Event();
        client.events.set(event.name, event);
        client.on(event.name, event.run.bind(event, client));
      }
    }
  }
}

module.exports = { 
  registerCommands, 
  registerEvents,
};"
`;

exports[`Templates should match getRegistryFileTS 1`] = `
"
import path from 'path';
import { promises as fs } from 'fs';
import DiscordClient from '../client/client';

export async function registerCommands(client: DiscordClient, dir: string = '') {
  const filePath = path.join(__dirname, dir);
  const files = await fs.readdir(filePath);
  for (const file of files) {
    const stat = await fs.lstat(path.join(filePath, file));
    if (stat.isDirectory()) registerCommands(client, path.join(dir, file));
    if (file.endsWith('.js') || file.endsWith('.ts')) {
      const { default: Command } = await import(path.join(dir, file));
      const command = new Command();
      client.commands.set(command.getName(), command);
      command.getAliases().forEach((alias: string) => {
        client.commands.set(alias, command);
      });
    }
  }
}

export async function registerEvents(client: DiscordClient, dir: string = '') {
  const filePath = path.join(__dirname, dir);
  const files = await fs.readdir(filePath);
  for (const file of files) {
    const stat = await fs.lstat(path.join(filePath, file));
    if (stat.isDirectory()) registerEvents(client, path.join(dir, file));
    if (file.endsWith('.js') || file.endsWith('.ts')) {
      const { default: Event } = await import(path.join(dir, file));
      const event = new Event();
      client.events.set(event.getName(), event);
      client.on(event.getName(), event.run.bind(event, client));
    }
  }
}
"
`;

exports[`Templates should match getTestCommand 1`] = `
"const BaseCommand = require('../../utils/structures/BaseCommand');

module.exports = class TestCommand extends BaseCommand {
  constructor() {
    super('test', 'testing', []);
  }

  async run(client, message, args) {
    message.channel.send('Test command works');
  }
}"
`;

exports[`Templates should match getTestCommandTS 1`] = `
"import { Message } from 'discord.js';
import BaseCommand from '../../utils/structures/BaseCommand';
import DiscordClient from '../../client/client';

export default class TestCommand extends BaseCommand {
  constructor() {
    super('test', 'testing', []);
  }

  async run(client: DiscordClient, message: Message, args: Array<string>) {
    message.channel.send('Test command works');
  }
}"
`;

exports[`Templates should match getTypescriptBotFile 1`] = `
"import { Client, ClientOptions, Collection } from 'discord.js';
import BaseEvent from '../utils/structures/BaseEvent';
import BaseCommand from '../utils/structures/BaseCommand';

export default class DiscordClient extends Client {

  private _commands = new Collection<string, BaseCommand>();
  private _events = new Collection<string, BaseEvent>();
  private _prefix: string = '!';

  constructor(options?: ClientOptions) {
    super(options);
  }

  get commands(): Collection<string, BaseCommand> { return this._commands; }
  get events(): Collection<string, BaseEvent> { return this._events; }
  get prefix(): string { return this._prefix; }

  set prefix(prefix: string) { this._prefix = prefix; }

}
"
`;
